docs/designs/high-level.md

```diff
@@ -1,33 +1,108 @@
+          direction TB
+          opensearch-reconciler[/OpenSearch Reconciler function/] --Create & execute -->reconcilers
+          subgraph reconcilers [Reconcilers Chain]
+              reconciler1[Reconciler 1] --- reconciler2
+              reconciler2[Reconciler 2] --- reconciler3
+              reconciler3[Reconciler 3] --- reconciler_etc[...]
+          end
+          subgraph components
+              scaler([Scaler component])
+              auto-scaler([Auto-Scaler Component]) 
+          end
+      end
+      
+      controller-- watches -->cr{{OpenSearchCluster Custom Resource}}
+      
+      subgraph kubernetes-objects-created [Created K8s Objects]
+          direction LR
+          opensearch-cluster-statefulsets[OpenSearch Cluster Statefulsets]
+          openseach-config-map[OpenSearch ConfigMap]
+          opensearch-service[OpenSearch Service]
+          ...
+      end    
+
+      subgraph monitoring [Monitoring Tools]
+          prometheus[Prometheus]
+          grafana[Grafana]
+          alert-manager[Alert Manager]
+      end
+      
+      manager --> kubernetes-objects-created
+```
+
+### Provisioning 
+The operator contains a Manager, which manges a Controller for its Custom Resource (`OpenSearchCluster`). It is mainly a listener on queue, which gets filled when ever a change has been made to one of the custom resources. The action is adjusting the deployed cluster to match the desired configuration specified in the custom resource. It can be a simple change to the Security configuration which may only update dynamic settings using OpenSearch API,or it can also be a change to the pod spec, increasing memory requirements from 10gb to 12gb. The latter requires a series of steps to accomplish that.
+
+Once a change has been detected it activates the Controller with the changed custom resource ID, which activates the OpenSearch Reconciler reconcile function. That reconciler is the main entry for the whole operator. Its goal is to modify/create/delete the OpenSearch Cluster and other deployed tools (e.g. Prometheus) reflecting the changes made to the custom resource. 
+
+The OpenSearch Reconciler reconcile function is primarily composed of two types of elements: Reconcilers and Components. The Reconcilers main job is to update/create the deployed Kubernetes objects representing the OpenSearch Cluster and other various software the Operator needs. If we take the Custom Resource YAML and split into it logical parts (i.e. fields), then a Reconciler is mapped to a specific part (field). The operators have the following Reconcilers:
+* Cluster Reconciler: mapped to `nodePools`. In charge of OpenSearch cluster
+* Dashboards Reconciler: mapped to `dashboards`. In charge of [OpenSearch Dashboards](https://opensearch.org/docs/latest/dashboards/index/) 
+* Configuration Reconciler - In charge of OpenSearch configuration
+* TLS Reconciler - mapped to `security.tls`. In charge of TLS configuration (OpenSearch Security Plugin)
+
```

Comment:
Missing here: Securityconfig Reconciler - mapped to `security.config`. In charge of the securityconfig with users and roles (OpenSearch Security plugin)